superset/views/core.py

```diff
@@ -2613,6 +2606,15 @@ def sql_json(self):
             raise Exception(_("Query record was not created as expected."))
         logging.info("Triggering query_id: {}".format(query_id))
 
+        rejected_tables = security_manager.rejected_tables(sql, mydb, schema)
+        if rejected_tables:
+            return json_error_response(
```

Comment:
We should probably set the query status to `FAILED` here with an error message, do a `session.commit()` then return the error.